mesonbuild/cmake/interpreter.py

```diff
@@ -827,11 +831,15 @@ def configure(self, extra_cmake_options: T.List[str]) -> CMakeExecutor:
         cmake_args += cmake_get_generator_args(self.env)
         cmake_args += [f'-DCMAKE_INSTALL_PREFIX={self.install_prefix}']
         cmake_args += extra_cmake_options
-        if not any(arg.startswith('-DCMAKE_BUILD_TYPE=') for arg in cmake_args):
-            # Our build type is favored over any CMAKE_BUILD_TYPE environment variable
-            buildtype = T.cast('str', self.env.coredata.get_option(OptionKey('buildtype')))
-            if buildtype in buildtype_map:
-                cmake_args += [f'-DCMAKE_BUILD_TYPE={buildtype_map[buildtype]}']
+        if any(arg.startswith('-DCMAKE_BUILD_TYPE=') for arg in cmake_args):
+            # Allow to override the CMAKE_BUILD_TYPE environment variable
+            mlog.debug('CMake build type explicitly set')
+            cmake_build_type = next(arg for arg in cmake_args if arg.startswith('-DCMAKE_BUILD_TYPE=')).split('=')[1]
+            self.build_type = buildtype_map[cmake_build_type] if cmake_build_type in buildtype_map else cmake_build_type
+        else:
+            mlog.debug('CMake build type set from the meson build type')
+            cmake_args += [f'-DCMAKE_BUILD_TYPE={self.build_type}']
```

Comment:
Wouldn't this set `CMAKE_BUILD_TYPE` to "`plain`" in case of an unsupported build type? `self.build_type` falls back to the Meson build type if it isn't in the map.

Some CMake projects have a default build type they use in case no build type is provided. I don't think we should break that by setting some weird `CMAKE_BUILD_TYPE`.

Reply:
@jpakkane determines the vision of Meson, but if it were me, I would greatly prefer to interfere as little as possible with external tools (i.e. CMake). I would trust the subproject developers in setting the appropriate defaults for their projects more than any blanket approach implemented in Meson. Put differently, I prefer Meson to implement the simplest logic possible to enable correct building (but not any simpler). The proposed build type logic feels both too invasive and too fragile to me.